recipes/mpfun90/meta.yaml

```
@@ -0,0 +1,83 @@
+{% set name = "mpfun90" %}
+{% set version = "2024-12-05" | replace('-', '.') %}
+
+package:
+  name: {{ name }}
+  version: {{ version }}
+
+source:
+  # need to match the GitHub version tag but only want one variable
+  url: https://github.com/APN-Pucky/mpfun90/archive/{{ version | replace('.', '-') }}.tar.gz
+  sha256: a273027f373eaa23743503612de4ca448628b5c1521510c21d5da3a5cfde79f5
+
+build:
+  skip: true  # [win]
+  number: 0
+  run_exports:
+    # MPFUN90 is archival software and the ABI should be static
+    - {{ pin_subpackage('mpfun90', max_pin='x') }}
+  ignore_run_exports_from:
+    - sed  # [not linux]
+  script:
+    # FIXME: https://github.com/APN-Pucky/mpfun90/commit/d00bfe06b8f69329464906817f7f8b795bed5388
+    - sed -i 's/ -Wl,--as-needed -Wl,-soname,libmpfun90.so / /g' Makefile  # [not linux]
+
+    - sed -i "s/libmpfun90.so/libmpfun90${SHLIB_EXT}/g" Makefile  # [not linux]
+    - export LDFLAGS="$LDFLAGS -lgfortran"  # [not linux]
+    - make dynamic FC="$FC" FFLAGS="$FFLAGS"
+    - mv libmpfun90${SHLIB_EXT} $PREFIX/lib/
+    - mkdir -p $PREFIX/include/mpfun90
+    - mv *.mod $PREFIX/include/mpfun90/
+    - make clean
+
+requirements:
+  build:
+    - {{ stdlib('c') }}
+    - {{ compiler('cxx') }}
+    - {{ compiler('fortran') }}
+    - make
+    - sed  # [not linux]
+
+test:
+  commands:
+    - test -f $PREFIX/lib/libmpfun90${SHLIB_EXT}
+    - test -f $PREFIX/include/mpfun90/mpfuna.mod
+    - test -f $PREFIX/include/mpfun90/mpfunb.mod
+    - test -f $PREFIX/include/mpfun90/mpfunc.mod
+    - test -f $PREFIX/include/mpfun90/mpfund.mod
+    - test -f $PREFIX/include/mpfun90/mpfune.mod
+    - test -f $PREFIX/include/mpfun90/mpfunf.mod
+    - test -f $PREFIX/include/mpfun90/mpfung.mod
+    - test -f $PREFIX/include/mpfun90/mpfunh.mod
+    - test -f $PREFIX/include/mpfun90/mpfuni.mod
+    - test -f $PREFIX/include/mpfun90/mpfunj.mod
+    - test -f $PREFIX/include/mpfun90/mpfunmod.mod
+    - test -f $PREFIX/include/mpfun90/mpdefmod.mod
+    - test -f $PREFIX/include/mpfun90/mpintmod.mod
+    - test -f $PREFIX/include/mpfun90/mprealmod.mod
+    - test -f $PREFIX/include/mpfun90/mpcmpmod.mod
+    - test -f $PREFIX/include/mpfun90/mpgenmod.mod
+    - test -f $PREFIX/include/mpfun90/mpfunsubmod.mod
+    - test -f $PREFIX/include/mpfun90/mpmodule.mod
+    - test -f $PREFIX/include/mpfun90/mpmodulem.mod
+    - test -f $PREFIX/include/mpfun90/mpmodulex.mod
+
+about:
+  home: https://github.com/APN-Pucky/mpfun90
+  summary: 'MPFUN90: A multiple precision floating point computation package'
+  description: |
+    MPFUN90 is software developed by David H. Bailey.
+    It is based on the work described in
+    David H. Bailey, "_A Fortran 90-based multiprecision system_,"
+    ACM Transactions on Mathematical Software (TOMS), 21 (1995) 379
+    DOI: [10.1145/212066.212075](https://doi.org/10.1145/212066.212075)
+  # modified BSD-3-Clause-LBNL license
+  license: DHB
```

The linter is correctly flagging this as not an SPDX license. All of David H. Bailey's software (https://www.davidhbailey.com/dhbsoftware/) is distributed under his DHB license (https://www.davidhbailey.com/dhbsoftware/DHB-License.txt) which is a modified version of the [BSD-3-Clause-LBNL license](https://spdx.org/licenses/BSD-3-Clause-LBNL.html) in which the author and copyright holder is himself and not the University of California, Lawrence Berkeley National Laboratory, or the United States government.

I'm happy to try to answer as best I can if there are questions on this.

---

You can make it SPDX
```suggestion
  # modified BSD-3-Clause-LBNL license
  license: LicenseRef-DHB
```

---

For future me: This works because of the following in [SPDX: Annex D SPDX license expressions (Normative)](https://spdx.github.io/spdx-spec/v2.2.2/SPDX-license-expressions/):

> SPDX License Expressions provide a way for one to construct expressions that more accurately represent the licensing terms typically found in open source software source code. A license expression could be a single license identifier found on the SPDX License List; a user defined license reference denoted by the LicenseRef-`[idString]`; a license identifier combined with an SPDX exception; or some combination of license identifiers, license references and exceptions constructed using a small set of defined operators (e.g., `AND`, `OR`, `WITH` and `+`).